src/simcore_service_deployment_agent/auto_deploy_task.py

```diff
@@ -341,17 +360,41 @@ async def _deploy(
 ) -> DockerRegistriesWatcher:
     app_config = app[APP_CONFIG_KEY]
     app_session = app[TASK_SESSION_NAME]
-    log.info("checking for changes...")
+
+    log.info("check if stacks exist...")
+    stacks_exist = await do_portainer_stacks_exist(app_config, app_session)
+    if not stacks_exist:
+        log.warning("stacks do not exist, initialising...")
+        # notifications
+        stack_cfg = await create_stack(git_task, app_config)
+        await deploy_portainer_stacks(app_config, app_session, stack_cfg)
+        await notify(
+            app_config,
+            app_session,
+            message="Stack was not found and re-initialised.",
+        )
+        main_repo = app_config["main"]["docker_stack_recipe"]["workdir"]
+        await notify_state(
+            app_config,
+            app_session,
+            state=app["state"][TASK_NAME],
+            message="Stack was not found and re-initialised.",
+        )
+        log.info("initialisation completed")
+
+    log.info("Checking for changes...")
     changes = await check_changes([git_task, docker_task])
     if not changes:
+        log.info("--> no changes detected")
```

Comment:
For me it is something I wanted to see on INFO when doing my dryruns with this (and the code of the following two PRs), it will print one message for every poll which by default happens every 60 seconds